x/protorev/keeper/rebalance.go

```diff
@@ -220,37 +222,43 @@ func (k Keeper) ExecuteTrade(ctx sdk.Context, route poolmanagertypes.SwapAmountI
 		return err
 	}
 
+	// Create and emit the backrun event and add it to the context
+	err = EmitBackrunEvent(ctx, pool, inputCoin, profit, tokenOutAmount, remainingTxPoolPoints, remainingBlockPoolPoints)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // RemainingPoolPointsForTx calculates the number of pool points that can be consumed in the current transaction.
-func (k Keeper) RemainingPoolPointsForTx(ctx sdk.Context) (uint64, error) {
-	maxRoutesPerTx, err := k.GetMaxPointsPerTx(ctx)
+func (k Keeper) GetRemainingPoolPoints(ctx sdk.Context) (uint64, uint64, error) {
+	maxPoolPointsPerTx, err := k.GetMaxPointsPerTx(ctx)
 	if err != nil {
-		return 0, err
+		return 0, 0, err
 	}
 
-	maxRoutesPerBlock, err := k.GetMaxPointsPerBlock(ctx)
+	maxPoolPointsPerBlock, err := k.GetMaxPointsPerBlock(ctx)
 	if err != nil {
-		return 0, err
+		return 0, 0, err
 	}
 
-	currentRouteCount, err := k.GetPointCountForBlock(ctx)
+	currentPoolPointsUsedForBlock, err := k.GetPointCountForBlock(ctx)
 	if err != nil {
-		return 0, err
+		return 0, 0, err
 	}
 
 	// Edge case where the number of routes consumed in the current block is greater than the max number of routes per block
 	// This should never happen, but we need to handle it just in case (deal with overflow)
-	if currentRouteCount >= maxRoutesPerBlock {
-		return 0, nil
+	if currentPoolPointsUsedForBlock >= maxPoolPointsPerBlock {
+		return 0, 0, nil
 	}
 
 	// Calculate the number of routes that can be iterated over
-	numberOfIterableRoutes := maxRoutesPerBlock - currentRouteCount
-	if numberOfIterableRoutes > maxRoutesPerTx {
-		return maxRoutesPerTx, nil
+	numberOfAvailablePoolPoints := maxPoolPointsPerBlock - currentPoolPointsUsedForBlock
+	if numberOfAvailablePoolPoints > maxPoolPointsPerTx {
+		return maxPoolPointsPerTx, numberOfAvailablePoolPoints, nil
 	}
 
-	return numberOfIterableRoutes, nil
+	return numberOfAvailablePoolPoints, numberOfAvailablePoolPoints, nil
```

Comment:
Could you please update godoc indicating what the return values are and when max vs available are returned?

Reply:
Updated in: https://github.com/osmosis-labs/osmosis/pull/4878/commits/03f8e989217c3839ed158bed02182f53cfada502